[DataGridPremium] AI Assistant (#16992)
mui/mui-x: docs/data/data-grid/ai-assistant/ai-assistant.md

<p class="description">Translate natural language into Data Grid views.</p>

:::warning
To use this feature you must have a prompt processing backend.
MUI [offers this service](/x/react-data-grid/ai-assistant/#with-muis-service) as a part of a premium package add-on.
Email us at [sales@mui.com](mailto:sales@mui.com) for more information.
:::

The AI Assistant feature lets users interact with the Data Grid component using natural language.
Type a prompt like "sort by name", "show amounts larger than 1000", or even make more complex queries like "which customers brought the most revenue the past year" in the prompt input field and the Data Grid will update accordingly.
In [supported browsers](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#browser_compatibility), users can also prompt the assistant using their voice.

To enable this feature on the Data Grid, pass the `aiAssistant` prop.

## Improving accuracy with example values

To increase the accuracy of the language processing, provide example values for the available columns via one of the methods below.

### Provide custom examples

> Use the `examples` prop in the `columns` array to provide custom examples as context for prompt processing.

Reviewer: ```suggestion
Use the `examples` property on items of the `columns` array to provide custom examples as context for prompt processing.
```

> The `examples` prop should contain an array of possible values for its respective column.

Reviewer: ```suggestion
The `examples` property should contain an array of possible values for its respective column.
```

---
title: Ask Your Table - AI Assistant
---

# Ask Your Table - AI Assistant [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

<p class="description">Translate natural language into Data Grid views.</p>

:::warning
To use this feature you must have a prompt processing backend.
MUI [offers this service](/x/react-data-grid/ai-assistant/#with-muis-service) as a part of a premium package add-on.
Email us at [sales@mui.com](mailto:sales@mui.com) for more information.
:::

The AI Assistant feature lets users interact with the Data Grid component using natural language.
Type a prompt like "sort by name", "show amounts larger than 1000", or even make more complex queries like "which customers brought the most revenue the past year" in the prompt input field and the Data Grid will update accordingly.
In [supported browsers](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#browser_compatibility), users can also prompt the assistant using their voice.

> To enable this feature on the Data Grid, pass the `aiAssistant` prop and use `<GridAiAssistantPanel />` in `aiAssistantPanel` slot.

Reviewer: ````suggestion
To enable this feature on the Data Grid, pass the `aiAssistant` prop and use the `GridAiAssistantPanel` component for `aiAssistantPanel` slot:

```tsx
import { DataGridPremium, GridAiAssistantPanel } from '@mui/x-data-grid-premium';
// ...
<DataGridPremium aiAssistant slots={{ aiAssistantPanel: GridAiAssistantPanel }} />
```
````

---
title: Ask Your Table - AI Assistant
---

# Ask Your Table - AI Assistant [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

<p class="description">Translate natural language into a set of grid state updates and apply them to the Data Grid component.</p>

:::warning
To use this feature you must have a prompt processing backend.
MUI [offers this service](/x/react-data-grid/ai-assistant/#with-muis-service) as a part of a premium package add-on.
Email us at [sales@mui.com](mailto:sales@mui.com) for more information.
:::

The AI assistant feature lets users interact with the Data Grid component using natural language.
Type a prompt like "sort by name" or "show amounts larger than 1000" in the prompt input field and the Data Grid will update accordingly.
In [supported browsers](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#browser_compatibility), users can also prompt the assistant using their voice.

> To enable client-side of this feature, pass `aiAssistant` prop.

Author: added as a follow-up

   :::success
   You can implement `onPrompt()` with `unstable_gridDefaultPromptResolver()`.
   This adds the necessary headers and stringifies the body in the correct format for you.

   It also makes it possible to provide additional context for better processing results, as shown below:

   ```ts
   const PROMPT_RESOLVER_PROXY_BASE_URL =
     process.env.NODE_ENV === 'development'
       ? 'http://localhost:3000'
       : 'https://api.my-proxy.com';

   function processPrompt(query: string, context: string) {
     const additionalContext = `The rows represent: List of employees with their company, position and start date`;

     return unstable_gridDefaultPromptResolver(
       `${PROMPT_RESOLVER_PROXY_BASE_URL}/api/my-custom-path`,
       query,
       context,
       additionalContext,
     );
   }
   ```

   :::

4. Provide data examples by either allowing data sampling with `allowAiAssistantDataSampling` prop or by filling the `examples` prop in the `columns` array.

### With a custom service

The Data Grid exposes several key elements of the AI Assistant feature so you can build your own prompt processing service:

> - The [AI Assistant Panel](/x/react-data-grid/components/ai-assistant-panel/) and [Prompt Field](/x/react-data-grid/components/prompt-field/) components can be used to build custom UI.

Author: Good point.

I have removed it from here, but didn't add the new sections since we still have to expose all pieces of the panel
https://deploy-preview-16992--material-ui-x.netlify.app/x/react-data-grid/components/ai-assistant-panel/

I see that other panel are also not mentioned in the feature pages
@KenanYusuf do we have an issue to have these components/pages updated

> - The [AI Assistant Panel](/x/react-data-grid/components/ai-assistant-panel/) and [Prompt Field](/x/react-data-grid/components/prompt-field/) components can be used to build custom UI.

Reviewer 2: It makes sense to add these sections once the rest of the components parts are built, otherwise it feels like false advertising 😅 . Agree with removing the references for now.

- The [`aiAssistant` API](/x/api/data-grid/grid-api/#grid-api-prop-aiAssistant) for processing the prompt results and updating state
- The `unstable_gridDefaultPromptResolver()` method for passing the prompt and context with the necessary headers to the processing service

Integrate these elements with your custom components and methods to suit your specific use case.

You can use a fully custom solution and apply the processing result using other Grid APIs such as [`setFilterModel()`](/x/api/data-grid/grid-api/#grid-api-prop-setFilterModel) or [`setSortModel()`](/x/api/data-grid/grid-api/#grid-api-prop-setSortModel) without the need to structure it as a `PromptResponse`.

> To replace `unstable_gridDefaultPromptResolver()` with your own solution, send a POST request to MUI's API.

Reviewer 3: very strange! Thanks for the heads up.

:::

The AI assistant feature lets users interact with the Data Grid component using natural language.
Type a prompt like "sort by name", "show amounts larger than 1000", or even make more complex queries like "which customers brought the most revenue the past year" in the prompt input field and the Data Grid will update accordingly.
In [supported browsers](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#browser_compatibility), users can also prompt the assistant using their voice.

To enable this feature on the Data Grid, pass the `aiAssistant` prop.

## Improving accuracy with example values

To increase the accuracy of the language processing, provide example values for the available columns via one of the methods below.

:::info
AI assistant demos use a utility function `mockPromptResolver()` to simulate the API that resolves user's prompts.
In a real-world scenario, you'd need to replace this with [MUI's processing service](/x/react-data-grid/ai-assistant/#with-muis-service) or [your own custom service](/x/react-data-grid/ai-assistant/#with-a-custom-service).

`mockPromptResolver()` can handle a predefined set of prompts:

> - `sort by name`

Reviewer 3: I don't think these should be in backticks since they're not lines of code.

- `sort by company name and employee name`
- `show people from the EU`
- `order companies by amount of people`

You can use suggestions to quickly enter prompts that are supported by the mock resolver.

:::

### Provide custom examples

Use the `examples` prop in the `columns` array to provide custom examples as context for prompt processing.
The `examples` prop should contain an array of possible values for its respective column.

{{"demo": "AssistantWithExamples.js", "bg": "inline"}}

### Use row data for examples

Pass the `allowAiAssistantDataSampling` prop to use row data to generate examples.
This is useful if you're dealing with non-sensitive data and want to skip creating custom examples for each column.

Data is collected randomly at the cell level, which means that the examples for a given column might not come from the same rows.

{{"demo": "AssistantWithDataSampling.js", "bg": "inline"}}

### Using server-side data

> The example below shows how to combine AI assistant with [server-side data](/x/react-data-grid/server-side-data/).

Reviewer 3: ```suggestion
The example below shows how to combine the AI assistant with [server-side data](/x/react-data-grid/server-side-data/).
```

   This adds a new button to the Toolbar that opens the Assistant Panel to receive the user's prompts.
3. Provide the `onPrompt()` callback to pass the user's prompts to the service.
   The service's response is used internally by the Data Grid to make the necessary state updates.

   :::success
   You can implement `onPrompt()` with `unstable_gridDefaultPromptResolver()`.
   This adds the necessary headers and stringifies the body in the correct format for you.

   It also makes it possible to provide additional context for better processing results, as shown below:

   ```ts
   const PROMPT_RESOLVER_PROXY_BASE_URL =
     process.env.NODE_ENV === 'development'
       ? 'http://localhost:3000'
       : 'https://api.my-proxy.com';

   function processPrompt(query: string, context: string) {
     const additionalContext = `The rows represent: List of employees with their company, position and start date`;

     return unstable_gridDefaultPromptResolver(
       `${PROMPT_RESOLVER_PROXY_BASE_URL}/api/my-custom-path`,
       query,
       context,
       additionalContext,
     );
   }
   ```

   :::

4. Provide data examples in either of the following ways:

>    - Fill the `examples` prop in the `columns` array; it's recommended if you want to avoid exposing the row data to the AI assistant.

Reviewer 3: This is a great use case for en dashes (–, `opt` + `-` on a Mac) 😁

```suggestion
   - Fill the `examples` prop in the `columns` array – this is recommended if you want to avoid exposing the row data to the AI assistant.
```

>    - Provide access to the row data with `allowAiAssistantDataSampling` prop; since actual data is used, it may result in better processing results.

Reviewer 3: ```suggestion
   - Provide access to the row data with `allowAiAssistantDataSampling` prop – since this uses real data, it may lead to better processing results.
```

### With a custom service

> The Data Grid exposes elements of the AI Assistant feature so you can build your own prompt processing service:

Reviewer 3: Do we want to capitalize AI Assistant or not? This doc does both but lowercase seems to be used more. I don't have a strong preference but it should be consistent everywhere. Is this the Proper Noun Name of the feature or is it Ask Your Table?

```suggestion
The Data Grid exposes elements of the AI assistant feature so you can build your own prompt processing service:
```

---
title: Ask Your Table - AI Assistant
---

# Ask Your Table - AI Assistant [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

<p class="description">Translate natural language into a set of grid state updates and apply them to the Data Grid component.</p>

:::warning
To use this feature you must have a prompt processing backend. 
MUI [offers this service](/x/react-data-grid/ai-assistant/#with-muis-service) as a part of a premium package add-on.

> Email us at [sales@mui.com](mailto:sales@mui.com) for more information.

Reviewer 4: I'm wondering if **Request a demo** pointing to some Google form would be a better channel for communication as we could have a template gathering pin-point expectations a user may have, requiring less to-and-fro communication 🤔

Author: CC @joserodolfofreitas

:::

The AI assistant feature lets users interact with the Data Grid component using natural language.
Type a prompt like "sort by name" or "show amounts larger than 1000" in the prompt input field and the Data Grid will update accordingly.

> In supported browsers, users can also prompt the assistant using their voice.

Reviewer 4: ```suggestion
In [supported browsers](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#browser_compatibility), users can also prompt the assistant using their voice.
```

This is useful if you're dealing with non-sensitive data and want to skip creating custom examples for each column.

Data is collected randomly at the cell level, which means that the examples for a given column might not come from the same rows.

For the `aiAssistant.getPromptContext()` API method, pass the `allowDataSampling` flag as a parameter.

```ts
const context = React.useMemo(
  () => apiRef.current.aiAssistant.getPromptContext(allowDataSampling),
  [apiRef, allowDataSampling],
);
```

{{"demo": "AssistantWithDataSampling.js", "bg": "inline"}}

## Using Server-side data

The example below shows how to combine prompt control with [server-side data](/x/react-data-grid/server-side-data/).

{{"demo": "AssistantWithDataSource.js", "bg": "inline"}}

## Processing service integration

Natural language prompts must be processed by a service to understand what kinds of state changes must be applied to the Data Grid to match the user's request.
You can use MUI's processing service or build your own.

### With MUI's service

The Data Grid provides all the necessary elements for integration with MUI's service.

> 1. Enable the AI Assistant feature by adding the `enableAiAssistant` prop.

Reviewer 4: I feel that from a user's pov, provisioning a backend comes earlier than integrating it with the client. Could we swap the order of points 1 and 2?

Update: Noticed Sam already commented a similar thing earlier 🙈

   A new toolbar button will appear in the default [`<Toolbar />`](/x/react-data-grid/components/toolbar/).
   This button opens `<AssistantPanel />` that can take user's prompts.
2. Contact [sales@mui.com](mailto:sales@mui.com) to get an API key for our processing service.

>    :::error

Reviewer 4: Nitpick: I perceive the red error callout with some limitation (something not achievable at all), could we replace it with a warning?
```suggestion
   :::warning
```

Author: I am fine with that as well.
Added it as an error because you can lose money if you don't do it right 🥲

   Avoid exposing the API key to the client by using a proxy server that receives prompt processing requests, adds the `x-api-key` header, and passes the request on to MUI's service.

   This is an example of a [Fastify proxy](https://www.npmjs.com/package/@fastify/http-proxy) for the prompt requests.

   ```ts
   fastify.register(proxy, {
     upstream: 'https://api.mui.com',
     prefix: '/api/my-custom-path',
     rewritePrefix: '/api/v1/datagrid/prompt',
     replyOptions: {
       rewriteRequestHeaders: (_, headers) => ({
         ...headers,
         'x-api-key': process.env.MUI_DATAGRID_API_KEY,
       }),
     },
   });
   ```

   :::

> 3. Provide the `onPrompt()` callback to pass the user's prompts to the service.

Reviewer 4: I assume that an end-to-end example isn’t feasible in this case. Given that, I think it would be helpful for users to have a basic starter template repo that includes the proxy setup, allowing them to easily clone and integrate it with their backend of choice.

Would it make sense to consider this as a potential follow-up action for this PR?

Reviewer 5: Yeah, we can evaluate the effort on a followup. 
Now that we have the model working for the examples, users can play with it already.

---
title: Ask Your Table - AI Assistant
---

# Ask Your Table - AI Assistant [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

<p class="description">Translate natural language into a set of grid state updates and apply them to the Data Grid component.</p>

:::warning
To use this feature, you need to have a prompt processing backend. MUI offers this [service](/x/react-data-grid/ai-assistant/#with-muis-service) as a part of a premium package add-on.
Email us at [sales@mui.com](mailto:sales@mui.com) to get more information.
:::

The AI assistant feature allows users to interact with the Data Grid component using natural language.
Type the prompt like "sort by name" or "show amounts larger than 1000" in the prompt input field, and the Data Grid will update accordingly.

> To enable client-side of this feature, pass `enableAiAssistant` prop.

Author: Updated with the suggestion https://github.com/mui/mui-x/pull/16992#discussion_r2028761217

For `aiAssistant.getPromptContext()` API method, pass `allowDataSampling` flag as a parameter.

```ts
const context = React.useMemo(
  () => apiRef.current.aiAssistant.getPromptContext(allowDataSampling),
  [apiRef, allowDataSampling],
);
```

{{"demo": "AssistantWithDataSampling.js", "bg": "inline"}}

## Using Server-side data

An example of combining prompt control with the [Server-side data](/x/react-data-grid/server-side-data/)

{{"demo": "AssistantWithDataSource.js", "bg": "inline"}}

## Processing service integration

Natural language prompts must be processed by a service to understand what kind of state changes must be applied to the Data Grid to match the user's request.
You can use MUI's processing service or build it by yourself.

### With MUI's service

Data Grid provides all necessary pieces to make the service integration easy.

1. Enable the AI Assistant feature by adding `enableAiAssistant` prop.
   A new toolbar button will appear in the default [`<Toolbar />`](/x/react-data-grid/components/toolbar/).
   This button opens `<AssistantPanel />` that can take user's prompts.

> 2. Contact [sales@mui.com](mailto:sales@mui.com) to get an API key for our processing service.

Reviewer 3: Maybe this should be step 1? Seems like you'd want to take care of this before all else.

   :::error

>    Avoid exposing the API key to the client by using a proxy server that receives prompt processing requests, adds the `x-api-key` header, and passes the request further to the MUI's service.

Reviewer 3: ```suggestion
   Avoid exposing the API key to the client by using a proxy server that receives prompt processing requests, adds the `x-api-key` header, and passes the request on to MUI's service.
```

   This is an example of a [Fastify proxy](https://www.npmjs.com/package/@fastify/http-proxy) for the prompt requests.

   ```ts
   fastify.register(proxy, {
     upstream: 'https://api.mui.com',
     prefix: '/api/my-custom-path',
     rewritePrefix: '/api/v1/datagrid/prompt',
     replyOptions: {
       rewriteRequestHeaders: (_, headers) => ({
         ...headers,
         'x-api-key': process.env.MUI_DATAGRID_API_KEY,
       }),
     },
   });
   ```

   :::

> 3. Provide `onPrompt()` callback to pass the user's prompts to the service.

Reviewer 3: ```suggestion
3. Provide the `onPrompt()` callback to pass the user's prompts to the service.
```

>    Service's response will be used internally to make the necessary state updates.

Reviewer 3: ```suggestion
   The service's response is used internally by the Data Grid to make the necessary state updates.
```

   :::success
   You can implement `onPrompt()` with `unstable_gridDefaultPromptResolver()`.

>    It will add necessary headers and stringify the body in the right format for you.

Reviewer 3: ```suggestion
   This adds the necessary headers and stringifies the body in the correct format for you.
```

>    In addition to this, it allows you to add an additional prompt context to achieve better processing results.

Reviewer 3: ```suggestion
  It also makes it possible to provide additional context for better processing results, as shown below:
```

   ```ts
   const PROMPT_RESOLVER_PROXY_BASE_URL =
     process.env.NODE_ENV === 'development'
       ? 'http://localhost:3000'
       : 'https://api.my-proxy.com';

   function processPrompt(query: string, context: string) {
     const additionalContext = `The rows represent: List of employees with their company, position and start date`;

     return unstable_gridDefaultPromptResolver(
       `${PROMPT_RESOLVER_PROXY_BASE_URL}/api/my-custom-path`,
       query,
       context,
       additionalContext,
     );
   }
   ```

   :::

> ### With custom service

Reviewer 3: ```suggestion
### With a custom service
```

> Use pieces of the AI Assistant feature to build your own prompt processing service.

Reviewer 3: ```suggestion
The Data Grid exposes several key elements of the AI Assistant feature so you can build your own prompt processing service:
```

> - `<GridAiAssistantPanel />` and `<GridPromptField />` components can be used to build custom UI.

Reviewer 3: ```suggestion
- The `<GridAiAssistantPanel />` and `<GridPromptField />` components for custom UI
```

> - `<GridAiAssistantPanel />` and `<GridPromptField />` components can be used to build custom UI.

Reviewer 2: I think we should link to the composable components here instead:

```suggestion
- The [AI Assistant Panel](/x/react-data-grid/components/ai-assistant-panel/) and [Prompt Field](/x/react-data-grid/components/prompt-field/) components can be used to build custom UI.
```

Reviewer 3: good call!

> - [`aiAssistant`](/x/api/data-grid/grid-api/#grid-api-prop-aiAssistant) API can be used to build the context using `getPromptContext()` and to apply the processing with `applyPromptResult(response: PromptResponse)` methods.

Reviewer 3: ```suggestion
- The [`aiAssistant` API](/x/api/data-grid/grid-api/#grid-api-prop-aiAssistant) for building context using `getPromptContext()`, and for applying processing with the `applyPromptResult(response: PromptResponse)` method
```

> - `unstable_gridDefaultPromptResolver()` can be used to pass the prompt and the context(s) with the necessary headers to the processing service.

Reviewer 3: ```suggestion
- The `unstable_gridDefaultPromptResolver()` method for passing the prompt and context with the necessary headers to the processing service
```

> Mix and match these with your custom components/methods to implement the processing the way you need it in your project.

Reviewer 3: ```suggestion
Integrate these elements with your custom components and methods to suit your specific use case.
```

> You can use completely custom solution and apply the processing result using other Grid's APIs like [`setFilterModel()`](/x/api/data-grid/grid-api/#grid-api-prop-setFilterModel) or [`setSortModel()`](/x/api/data-grid/grid-api/#grid-api-prop-setSortModel) without a need to structure it as `PromptResponse`.

Reviewer 3: ```suggestion
You can use a fully custom solution and apply the processing result using other Grid APIs such as [`setFilterModel()`](/x/api/data-grid/grid-api/#grid-api-prop-setFilterModel) or [`setSortModel()`](/x/api/data-grid/grid-api/#grid-api-prop-setSortModel) without the need to structure it as a `PromptResponse`.
```

> To replace `unstable_gridDefaultPromptResolver()` with your own solution, send a POST request to MUI's API with a library of your choice.

Reviewer 3: ```suggestion
To replace `unstable_gridDefaultPromptResolver()` with your own solution, send a POST request to MUI's API.
```

> Body of the request requires `query` and `context` parameters.

Reviewer 3: ```suggestion
The body of the request requires `query` and `context` parameters.
```

`additionalContext` is optional.

> API response type is `Result<PromptResponse>`.

Reviewer 3: ```suggestion
The API response type is `Result<PromptResponse>`.
```

---
title: Ask Your Table - AI Assistant
---

# Ask Your Table - AI Assistant [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan 'Premium plan')

<p class="description">Translate natural language into Data Grid views.</p>

:::warning
To use this feature you must have a prompt processing backend.
MUI [offers this service](/x/react-data-grid/ai-assistant/#with-muis-service) as a part of a premium package add-on.
Email us at [sales@mui.com](mailto:sales@mui.com) for more information.
:::

The AI assistant feature lets users interact with the Data Grid component using natural language.
Type a prompt like "sort by name", "show amounts larger than 1000", or even make more complex queries like "which customers brought the most revenue the past year" in the prompt input field and the Data Grid will update accordingly.
In [supported browsers](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition#browser_compatibility), users can also prompt the assistant using their voice.

To enable this feature on the Data Grid, pass the `aiAssistant` prop.

:::info

> To check out this feature without a processing backend, use `mockPromptResolver()` to simulate the API that resolves user's prompts.

Author: @joserodolfofreitas I am not sure if I should leave this or not

Reviewer 5: +1 to remove it. Let's not add another public API  to maintain later.

Reviewer 4: I’m wondering if this could be useful as a fallback when tokens are overused, or if the backend is unavailable for whatever reason. 🤷

Reviewer 2: We should keep it for development purposes, regardless of whether we document it.

Author: I will remove the info, but keep it in the code (it is part of the generator package)

- show people from the EU
- order companies by amount of people

You can use suggestions to quickly enter prompts that are supported by the mock resolver.

```ts
import { DataGridPremium } from '@mui/x-data-grid-premium';
import { mockPromptResolver } from '@mui/x-data-grid-generator';

...
  <DataGridPremium
    {...otherProps}
    aiAssistantSuggestions={[
      { value: 'Sort by name' },
      { value: 'Show people from the EU' },
      { value: 'Sort by company name and employee name' },
      { value: 'Order companies by amount of people' },
    ]}
    aiAssistant
    onPrompt={mockPromptResolver}
    showToolbar
  />
...

```

:::

## Improving accuracy with example values

To increase the accuracy of the language processing, provide example values for the available columns via one of the methods below.

### Provide custom examples

Use the `examples` prop in the `columns` array to provide custom examples as context for prompt processing.
The `examples` prop should contain an array of possible values for its respective column.

:::info
AI assistant demos use limited [MUI's processing service](/x/react-data-grid/ai-assistant/#with-muis-service).
:::

> {{"demo": "AssistantWithExamples.js", "bg": "inline"}}

Reviewer 5: I noticed on this demo that when the model uses "Id", for example to aggregate employees when grouping companies, the model doesn't make it visible, if it's invisible. Which feels like the change wasn't applied if users don't see it. In my example, it's correctly applied if you manually turn the visibility of the field on, but I was confused with the answer at first without seeing the column used for aggregation and sorting.

step 1
<img width="836" alt="image" src="https://github.com/user-attachments/assets/8d3b0066-e951-4eeb-ba1b-80b729e63133" />
step 2
<img width="836" alt="image" src="https://github.com/user-attachments/assets/edd748ec-8993-4de0-8a27-b6e132c1535b" />

Reviewer 5: (On this example it's not so bad, I realized that previously the group size was also hidden due to the size of the name of the first companies.)

Reviewer 2: Sounds like we need to factor in column visibility in the prompt response. I think this could come in a follow-up as it's kind of an edge case

Author: Added it here
https://github.com/mui/mui-x/issues/17292

Reviewer 5: Right, but maybe for the demo, we leave the Id column visible then?

Reviewer 2: > Right, but maybe for the demo, we leave the Id column visible then?

We could make that argument for all of the hidden columns in these demos, right? Should we make them all visible in the meantime?

Reviewer 5: > We could make that argument for all of the hidden columns in these demos, right? Should we make them all visible in the meantime?

I thought about the id, because any aggregating employees, would likely use the `id` field, but you're right, we should consider all the others.
I'd be +1 to have everything visible on the docs until we have the follow-up or remove the field from the Data Grid so the model doesn't use it.
